dds/DCPS/DataWriterImpl_T.h

```diff
@@ -469,7 +470,11 @@ DataWriterImpl_T
           TraitsType::type_name()));
         return 0;
       }
-
+      if (encapsulated && (encoding.kind() == Encoding::KIND_XCDR1
+        || encoding.kind() == Encoding::KIND_XCDR2)) {
```

Comment:
```suggestion
      if (encapsulated) {
```

The spec doesn't make this conditional on a specific encoding.

Reply:
This is what it says:
"The RTPS specification does not define any settings for the 2 byte options field and further states
that a receiver should not interpret it when it reads the options field. This DDS-XTYPES
specification changes this defining the use of some bits in the options field"
So I assumed it is 'XTYPES' specific. Is this incorrect interpretation?

Reply:
Just to clarify: the third 'kind' is:
    /**
     * This is the classic encoding of OpenDDS used when there is no RTPS
     * transport being used. It has no padding bytes and no XCDR behavior.
     */
    KIND_UNALIGNED_CDR,

So it looked to me these rules would not be applicable to it, but I might be wrong.

Reply:
> So I assumed it is 'XTYPES' specific. Is this incorrect interpretation?

It doesn't say any where it's just limited to the encoding defined by XTypes, it just says the XTypes spec is changing the definition of the options field.

> So it looked to me these rules would not be applicable to it, but I might be wrong.

They do not, but unaligned cdr isn't encapsulated and since the change is for all encapsulated encodings, it doesn't make sense to add any more conditions unto it.